### DISCUSSION TITLE
Is a code signing certificate required when developing for iOS locally?

### DISCUSSION THREAD

**Discussion by StefanVDWeide** [tauri-apps/tauri#6234] [Q&A]
I wanted to check out the new mobile app development capabilities in the 2.0 alpha release, but when running `cargo tauri ios init` I get the following error:

```
Error No code signing certificates found. You must add one and set the certificate development team ID on the `tauri > bundle > iOS > developmentTeam` config value or the `TAURI_APPLE_DEVELOPMENT_TEAM` environment variable. To list the available certificates, run `tauri info`.
```

When checking the documentation I end up on the [code signing page for MacOs](https://tauri.app/v1/guides/distribution/sign-macos/) which seems to relate to building and publishing applications rather than developing locally. Is there a similar page describing the necessary steps for developing an iOS app locally? Is a certificate even needed for development? A way around this would be great since Apple charges €99,- per year for a developer account.

Thanks!

**Comment by edwincoronado** [comment]
I just gave it a bogus value and got `pnpm tauri ios init` to work fine:

```sh
TAURI_APPLE_DEVELOPMENT_TEAM=12345 pnpm tauri ios init
```

**Comment by veryspry** [comment]
> I just gave it a bogus value and got pnpm tauri ios init to work fine:


Same worked for me 😅

**Comment by GLASFLEX** [comment]
Doesn't work anymore seems like the bogus method was patched

**Reply by InSuperposition** [reply]
In a recent version of `tauri` the variable was changed from 
`TAURI_APPLE_DEVELOPMENT_TEAM` to `APPLE_DEVELOPMENT_TEAM`

**Comment by actuallylost** [comment]
Do we know if we need to `export` the value (or add it in a `.env` file), put it in `Cargo.toml` or somewhere in `tauri.conf.json`?

**Reply by noruya** [reply]
U can write value to "tauri.conf.json" like this ->
"tauri": { "bundle": { "iOS": { "developmentTeam": "XXXXXXX"}}}

**Comment by Shyam-Chen** [comment]
Is it necessary to have a code signing certificate when running the `tauri ios build` command locally?

```json
{
  ...
  "identifier": "shyam.chen.dev",
  ...
  "bundle": {
    "iOS": {
      "developmentTeam": "Shyam Chen (Personal Team)"
    },
  ...
}
```

```sh
$ pnpm tauri ios build

> tauri "ios" "build"
> error: No account for team "Shyam Chen (Personal Team)". Add a new account in the Accounts preference pane or verify that your accounts have valid credentials.
> error: No profiles for 'shyam.chen.my-app' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'shyam.chen.my-app'.
```

**Comment by allochi** [comment]
I had the same problem yesterday, so I did this:

1. open Xcode -> Preferences -> Accounts
2. at the panel "Apple IDs" click the "+" button to add an ID
3. choose "Apple ID" at the dialog
4. login to Apple account (not necessary developer account)
5. once the account is listed click "Manage Certificates" in the right side panel
6. click the "+" button and choose "Apple Development"
7. reinitialize "tauri" project using `cargo tauri ios init`
8. run the project `cargo tauri ios dev`

This just worked for me without editing any file or adding any environment variables. I hope it help you too. enjoy! 😃

**Reply by darrenbarklie** [reply]
I ran Steps 7 & 8 with `pnpm`, but otherwise this is exactly what I needed. Thanks.

**Comment by jamesporter** [comment]
Just trying out first 2.0.

The above didn't work; however if opened the `src-tauri/gen/apple/*.xcodeproj` file and setup a team in Xcode. Was then able to run `pnpm tauri ios dev`.
